### ISSUE TITLE
How do I check if field is required when it depends on condition?

### ISSUE THREAD

**Issue by Charlie91** [jquense/yup#1539]
I've found ways to check if field is required, like:

```
export const isFieldRequired = (validationSchema: AnyObjectSchema, field: string): boolean => {
  const fieldProperties = validationSchema.describe().fields[field];

  if (fieldProperties && 'tests' in fieldProperties) {
    return fieldProperties.tests.some(test => test.name === 'required');
  }
  return false;
};
```

but all of them don't work if field's validation rules depend on condition, like:

```
  name: Yup.string().when('email', (email?: string) => {
    const rules = Yup.string()
      .max(140);

    if (email) {
      return rules.required();
    }
    return rules;
  }),

```

**Comment by jquense** [comment]
use `describe()`

**Comment by Charlie91** [comment]
I did use it
`
  const fieldProperties = validationSchema.describe().fields[field];
`

**Comment by jquense** [comment]
you need to provide options to describe so that it can resolve the conditions, otherwise it doesn't have enough information to tell. Consult the source if the docs don't demonstrate it thanks!

**Comment by SimmeNilsson** [comment]
@Charlie91 I was looking for something similar and came across this issue.
It took some time but I got the simple case for conditional required to work by passing in the form values as an object to 
_.describe({ value: formValuesObject })_
I'm using vee-validate so in my case I just took them from form.values that I dug up with FormContextKey.
However, one needs to run yup 1.0.0 beta (@next) for it to work as in 0.32 (@latest) _describe_ ignores any option object passed to it.

Line 454:
https://github.com/jquense/yup/blame/master/src/object.ts

Line 444:
https://github.com/jquense/yup/blame/266f33a53e27f2405c2c3644574c5956b8a757ec/src/object.ts

Next step for me was then to support nested objects by traversing the description based on the name path.
I've gotten it to work partly.
However, I'm a little stuck on arrays with objects where one field is required depending on another field in the same object.
From what I can see the description seems to have only one representation of the object in the array even if there are multiple instances and in some instances the field might be required and in some not.
@jquense Is this a limitation of describe? Is there any way to get a description for the individual items in the array as result may vary?
